run/jobs/e2e_test.py

```diff
@@ -36,36 +36,61 @@
 @pytest.fixture
 def setup_job():
     # Build container image and run the job
-    subprocess.check_call(
-        [
-            "gcloud",
-            "builds",
-            "submit",
-            "--config",
-            "e2e_test_setup.yaml",
-            "--project",
-            PROJECT,
-            "--substitutions",
-            "_SERVICE=" + SERVICE + ",_VERSION=" + SUFFIX + ",_REGION=" + REGION,
-        ]
-    )
+    # Retry up to 3 times
+    built = False
+    for x in range(3):
+        try:
+            subprocess.check_call(
+                [
+                    "gcloud",
+                    "builds",
+                    "submit",
+                    "--config",
+                    "e2e_test_setup.yaml",
+                    "--project",
+                    PROJECT,
+                    "--substitutions",
+                    "_SERVICE=" + SERVICE + ",_VERSION=" + SUFFIX + ",_REGION=" + REGION,
+                ]
+            )
+            built = True
+        except Exception as e:
+            print(e)
+
+        if built:
+            break
+
+        # Linear backoff
+        time.sleep(x * 10)
 
     yield SERVICE
 
     # Clean up the test resource
-    subprocess.check_call(
-        [
-            "gcloud",
-            "builds",
-            "submit",
-            "--config",
-            "e2e_test_cleanup.yaml",
-            "--project",
-            PROJECT,
-            "--substitutions",
-            "_SERVICE=" + SERVICE + ",_VERSION=" + SUFFIX + ",_REGION=" + REGION,
-        ]
-    )
+    destroyed = False
+    for x in range(3):
```

Comment:
Same 3 concerns as above! (backoff, exception, and cleanup using finally)

Reply:
This step looks like the clean-up step for the previous one.

In any case - is a `finally` clause required if we're working with `backoff`?

Reply:
It's a bit tricky in this case actually - I'm not sure what would be the best approach if `teardown` were to throw errors. In other cases I'd suggest `finally` to clean up but fine to leave it as is here.